doc/releases/changelog-0.29.0.md

```diff
@@ -844,6 +844,12 @@
 * Ensure that a `QNode` does not return an empty iterable.
   [(#3769)](https://github.com/PennyLaneAI/pennylane/pull/3769)
 
+* The keyword arguments for `qml.equal` now are used when comparing the observables of a 
+  Measurement Process. The eigvals of measurements are only requested if both observables are `None`,
+  saving computational effort.  The string representation of a Measurement Process now includes the `_eigvals`
+  property if it is set. Only converts input to `qml.Hermitian` to a numpy array if the input is a list.
```

Comment:
This feels like 3 things in one bullet point to me - maybe they should be separate points?